platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,157 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[favor_composite_builds]]
+== Favor `build-logic` Composite Builds over `buildSrc`
+:keywords: structuring-builds,composite-builds,buildSrc
+
+You should setup a <<composite_builds.adoc#defining_composite_builds,Composite Build>> (often called an "included build") to hold your build logic—including any custom plugins, convention plugins, and other build-specific customizations.
+
+=== Explanation
+
+The preferred location for build logic is an included build (typically named `build-logic`), **not** in `buildSrc`.
+
+The automatically available `buildSrc` is great for rapid prototyping, but it comes with some subtle disadvantages:
+
+* There are classloader differences in how these 2 approaches behave that can be surprising; included builds are treated just like external dependencies, which is a simpler mental model.  Dependency resolution behaves subtly differently in `buildSrc`.
```

Comment:
Is it true that buildSrc tests will not be run automatically and the way to do it is the cumbersome `./gradlew -p buildSrc test` command?

Reply:
Yes, see https://github.com/gradle/gradle/pull/33081#discussion_r2042161861 for a test that demos this.